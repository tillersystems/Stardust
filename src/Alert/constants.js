export const STATUS_ICON_NAMES = {
  info: 'info',
  success: 'check-mark',
  error: 'exclamation-mark',
  warning: 'exclamation-mark',
};
