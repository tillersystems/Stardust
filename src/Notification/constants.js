export const ADD_NOTIFICATION = 'ADD_NOTIFICATION';
export const REMOVE_NOTIFICATION = 'REMOVE_NOTIFICATION';
export const UPDATE_NOTIFICATION = 'UPDATE_NOTIFICATION';
