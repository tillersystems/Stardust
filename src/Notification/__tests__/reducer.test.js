import notificationReducer from '../reducer';
import { ADD_NOTIFICATION, REMOVE_NOTIFICATION } from '../constants';

const previousState = {
  notificationsCount: 0,
  notifications: [],
  options: {
    autoDismiss: false,
    autoDismissTimeout: 1000,
    pauseOnHover: false,
  },
};

describe('<NotificationProvider />', () => {
  test('should throw an error if incorrect type is passed', () => {
    const type = 'INCORRECT_TYPE';
    expect(() => {
      notificationReducer(previousState, {
        type,
        options: {
          autoDismiss: true,
          autoDismissTimeout: 5000,
          pauseOnHover: true,
        },
      });
    }).toThrow(`Unhandled action type: ${type}`);
  });

  test('should handle ADD_NOTIFICATION', () => {
    const nextState = notificationReducer(previousState, {
      type: ADD_NOTIFICATION,
      component: jest.fn(),
      options: {
        autoDismiss: true,
        autoDismissTimeout: 5000,
        pauseOnHover: true,
      },
    });
    expect(nextState).toMatchSnapshot();
  });

  test('should handle REMOVE_NOTIFICATION', () => {
    const nextState = notificationReducer(previousState, {
      type: REMOVE_NOTIFICATION,
    });
    expect(nextState).toMatchSnapshot();
  });
});
