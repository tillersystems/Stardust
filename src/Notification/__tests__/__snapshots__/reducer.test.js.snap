// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationProvider /> should handle ADD_NOTIFICATION 1`] = `
Object {
  "notifications": Array [
    Object {
      "component": [MockFunction],
      "key": "notif-component-0",
      "options": Object {
        "autoDismiss": true,
        "autoDismissTimeout": 5000,
        "pauseOnHover": true,
      },
    },
  ],
  "notificationsCount": 1,
  "options": Object {
    "autoDismiss": false,
    "autoDismissTimeout": 1000,
    "pauseOnHover": false,
  },
}
`;

exports[`<NotificationProvider /> should handle REMOVE_NOTIFICATION 1`] = `
Object {
  "notifications": Array [],
  "notificationsCount": 0,
  "options": Object {
    "autoDismiss": false,
    "autoDismissTimeout": 1000,
    "pauseOnHover": false,
  },
}
`;
